### ISSUE TITLE
cannot rollback - no transaction is active

### ISSUE THREAD

**Issue by pakyo-pan** [tekartik/sqflite#240]
When I `hot restart`, I got the following errors:
![image](https://user-images.githubusercontent.com/13342613/60781301-65efbf80-a174-11e9-9ab0-b7b546b0c8ea.png)
Use example code as below and then `hot restart`:
```dart
import 'package:path/path.dart';
import 'package:sqflite/sqflite.dart';

void main() async {
  String path = await getDatabasesPath();

  await openDatabase(
    join(path, 'ccc.db'),
    // onCreate: _onCreate,
    version: 1,
  );
}
```
I think this error is caused by not closing the database.

**Comment by alextekartik** [comment]
Does it work (i.e. is it only a line display in the output or a real error that prevent your app from running)?. Indeed there is some special trick to support hot reload/restart otherwise if you were in a transaction it would simply hangs

**Comment by pakyo-pan** [comment]
I'm sorry for my English.
This error only happened when `hot restart` and didn't prevent my app from running. 
I am sure that the previous version of sqflite does not have this error.

**Comment by congjiujiu** [comment]
I have met the same error. Would it be fixed asap?

**Comment by alextekartik** [comment]
@congjiujiu Can you elaborate (output, code, Was it on iOS or Android) on the error you got? Can you confirm that it was en error? (or just a printed output and the application worked fine). Was it during hot restart and do you have a way to reproduce it consistently (i.e. what is the ASAP need here?).

**Comment by zigmo** [comment]
I'm not sure if this is related to the same problem. But I'm getting this exception after clicking on "restart" in VSCode:

```
PlatformException (PlatformException(sqlite_error, Cannot perform this operation because there is no current transaction., {arguments: [], sql: null}))
````

But it looks likes its working without any issues with the version **1.1.3**.

Sometime I can click on continue multiple times and it still works - but sometimes it just hang and I have to stop and start again.

````
if (!isOpen) {
      final SqfliteDatabase database = newDatabase(path);
      await database.doOpen(options);
      sqfliteDatabase = database;
    }
    return sqfliteDatabase;
````
this is where is hangs: **await database.doOpen(options);**

```
[√] Flutter (Channel stable, v1.7.8+hotfix.3, on Microsoft Windows [Version 10.0.18362.239], locale de-DE)
    • Flutter version 1.7.8+hotfix.3 
    • Framework revision b712a172f9 (11 days ago), 2019-07-09 13:14:38 -0700
    • Engine revision 54ad777fd2
    • Dart version 2.4.0

[√] VS Code, 64-bit edition (version 1.25.1)
    • VS Code at C:\Program Files\Microsoft VS Code
    • Flutter extension version 2.21.1

[√] Connected device (1 available)
    • Android SDK built for x86 • emulator-5554 • android-x86 • Android 9 (API 28) (emulator)

• No issues found!
```

**Comment by examango** [comment]
@alextekartik I have met the same error.

Error calling sqlite3_step (1: cannot rollback - no transaction is active) SQLITE_ERROR
DB Query: ROLLBACK
Unknown error finalizing or resetting statement (1: cannot rollback - no transaction is active)
DB Query: ROLLBACK

If hot restart, It always happen.

But the application work fine.

**Comment by alextekartik** [comment]
I don't know whether the issue happened for you on iOS or Android. I would be interested to know if this is fixed using sqflite 1.1.6+4 released today. Thanks!

**Comment by pakyo-pan** [comment]
The issue has been fixed, thank you very much!
